spec/statsd/timer_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'spec_helper'
+
+describe Datadog::Statsd::Timer do
+  subject do
+    described_class.new(interval) do
+      sleep callback_durations.next
+      call_times << Time.now
+    end
+  end
+  let(:call_times) { Queue.new }
+  let(:callback_durations) do
+    Enumerator.new do |y|
+      loop do
+        y << 0
+      end
+    end
+  end
+
+  describe '#start' do
+    let(:interval) { 0.001 }
+
+    # this callback_durations allows calls at an odd number of times
+    # to take almost no time, whereas calls at an even number of times
+    # to take at least `interval` seconds.
+    let(:callback_durations) do
+      Enumerator.new do |y|
+        loop do
+          [0, interval].each do |d|
+            y << d
+          end
+        end
+      end
+    end
+
+    after do
+      subject.stop
+    end
+
+    it 'starts the timer thread and calls the callback' do
+      expect do
+        subject.start
+      end.to change { Thread.list.size }.by(1)
+
+      # use timeout just in case call_times.pop waits forever
+      Timeout.timeout(1) do
+        # the first call is made after `interval` seconds
+        first_call_time = call_times.pop
+        # the second call is made `interval` seconds after the first call
+        # and it takes `interval` seconds before Time.now is called
+        second_call_time = call_times.pop
+        # the third call is made immediatelly after the second call
+        third_call_time = call_times.pop
+        expect(second_call_time - first_call_time).to be_within(0.001).of(interval * 2)
+        expect(third_call_time - second_call_time).to be_within(0.001).of(0)
```

Comment:
1ms is a _very_ small time in the typically-very-constrained world of CI, leading to intermittent failures.  Can we use 100ms here?

Reply:
I've increased the value https://github.com/DataDog/dogstatsd-ruby/pull/231/commits/47f6cdf9f87eb7e1f01cb0cf27e91777567d10d6